docs/example-library/meltano-annotations/ending-meltano.yml

```diff
@@ -13,7 +13,11 @@ plugins:
   - name: tap-gitlab
     variant: meltanolabs
     pip_url: git+https://github.com/MeltanoLabs/tap-gitlab.git
+    annotations:
+      example-annotation-plugins-extractors-tap-gitlab: {}
     config:
+      annotations:
+        example-annotation-plugins-extractors-tap-gitlab-config: {}
```

Comment:
(1 of 2): I would like to keep `annotations` out of `config`, here and below. Because a `config` value can be anything, it's not clear that `annotations` isn't the name of a setting. Also - I can't imagine any use case where the `config` level is interesting to add annotations, rather than annotating one layer above (on the plugin in this case).

What do you think?

Reply:
That makes sense. I can remove the annotations from the example, and from the schema for the configs, but unfortunately the current implementation would still remove them from the merged project files.

To avoid this I think we'll need something like what has been implemented in #7148 under `meltano.core.manifest.jsonschema`, which can tell us every location a particular reference within a jsonschema appears. In this case we'd limit `ProjectFiles.extract_annotations` to only extract those which are located where a ref to `#/$defs/annotations` appears in the schema.

Reply:
I'll echo what AJ said - annotations at the config level could be very confusing about whether it's a configurable setting or not.

Reply:
Resolved in 86ef0e0